src/common/actions/snap-builds.js

```diff
@@ -18,27 +20,38 @@ const REQUEST_POST_OPTIONS = {
   credentials: 'same-origin'
 };
 
-export function fetchBuildsSuccess(builds) {
+export function fetchBuildsSuccess(id, builds) {
   return {
     type: FETCH_BUILDS_SUCCESS,
-    payload: builds
+    payload: {
+      id,
+      builds
+    }
   };
 }
 
-export function fetchBuildsError(error) {
+export function fetchBuildsError(id, error) {
```

Comment:
Of course, if LP goes down we will likely get same error for all.

Keeping the statuses separate in the store at least makes sure they don't affect each other.
For parallel requests it makes sure at least that error from one repo will not affect success from the other. Or maybe more importantly - error from one repo is not overridden by success response from the other (that comes back later).

I'd prefer to stick to this working solution now and discuss it further when we have more time for that (and any refactoring that may be needed).